samplers/jaegerremote/sampler_remote.go

```diff
@@ -273,13 +273,16 @@ type httpSamplingStrategyFetcher struct {
 }
 
 func newHTTPSamplingStrategyFetcher(serverURL string) *httpSamplingStrategyFetcher {
-	customTransport := http.DefaultTransport.(*http.Transport).Clone()
-	customTransport.ResponseHeaderTimeout = defaultRemoteSamplingTimeout
+	var rt http.RoundTripper
+	if t, ok := http.DefaultTransport.(*http.Transport); ok {
+		rt = t.Clone()
+	}
```

Comment:
Since the timeout is no longer set on the transport, can we not just delete this part? Default transport will be used automatically by the client, would it not? What's the purpose of cloning it?

Reply:
This avoids races on `http.DefaultTransport` in tests.

We can either:
* remove this and the test that causes the race https://github.com/open-telemetry/opentelemetry-go-contrib/pull/4045/files#diff-643eabddd4dc6c7bed5f17028eae0306783f2346732f1ab135e3d8becb5aa766R596
* leave it as is

Reply:
@morus12 Remove the test. It does not really add any value.